File: java/0009_Palindrome_Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        } else if (x < 10) {
            return true;
        }

        int res = 0;
        int y = x;

        while (y > 0) {
            res = res * 10 + y % 10;
            y /= 10;
        }

        return res == x;
    }
}

File: kotlin/0009_Palindrome_Number.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        return when {
            x < 0 -> false
            x < 10 -> true
            else -> {
                var res = 0
                var y = x
                while (y > 0) {
                    res = res * 10 + y % 10
                    y /= 10
                }
                res == x
            }
        }
    }
}